### PR TITLE
adding coreos install media

### PR DESCRIPTION
**Pull request by heckj**
Adding ansible tasks to pull down and install CoreOS installation
media (since it's small) to provide an example OS installation flow
with RackHD. Maps to the location and flow from the install-coreos.ipxe
profile (in on-http) and the "Task.Os.Install.CoreOS" workflow task in
on-tasks.

### CODE REVIEW

File: example/roles/repos/tasks/main.yml [hunk 1]
```diff
@@ -67,3 +67,14 @@
    - monorail-efi32-snponly.efi
    - monorail-efi64-snponly.efi
    - undionly.kpxe
+
+- name: Mark coreOS directory for hosting coreos demonstration booting
```
Comment (benbp) [inline example/roles/repos/tasks/main.yml]: "CoreOS"

**Comment by benbp** [comment]
:+1: with minor spelling nit.

**Comment by DavidjohnBlodgett** [comment]
:+1:

**Comment by heckj** [comment]
(yes, I'll squash after I get done screwing around with this)

**Comment by heckj** [comment]
@DavidjohnBlodgett yeah, in general - when I was doing "stupid dev tricks", I wanted to add a remote for my own fork, and realized I was having to screw with github auth using HTTPS:// and my dual factor authentication thing, or I could forward the SSH credentials when I did a `vagrant ssh` to get into the VM. Not strictly needed for this effort, but nice to have.

**Comment by DavidjohnBlodgett** [comment]
@heckj cool thanks!

**Comment by DavidjohnBlodgett** [comment]
:+1:

**Comment by heckj** [comment]
@DavidjohnBlodgett I ran through the script with the change, and it worked through brilliantly...

File: example/README.md [hunk 2]
```diff
@@ -1,114 +1,176 @@
...
-# the monorail_rack setup script.
+    curl -H "Content-Type: application/json" \
+    -X PUT --data @samples/virtualbox_install_coreos.json \
+    http://localhost:9090/api/1.1/workflows
 
-# deployment variables
-pxe_count=1
-```
+We also include the sku definition. RackHD sku support includes a mechanism that
+will run a workflow when a SKU is identified. In this case, the example uses the
```
Comment (benbp) [inline example/README.md]: Inconsistent capitalization of 'SKU'.

Also, this language is a little misleading perhaps? "sku support includes a mechanism that will run a workflow when a SKU is identified." We don't run a workflow right when the SKU is identified, it has to be triggered by another workflow.

File: example/bin/monorail_rack [hunk 3]
```diff
@@ -61,16 +54,20 @@ if [ $pxe_count ]
...
-        vboxmanage modifyvm $vmName --nic1 intnet --intnet1 closednet --nicpromisc1 allow-all;
-        vboxmanage modifyvm $vmName --nictype1 82540EM --macaddress1 auto;
-
+        if [[ ! -e $vmName.vdi ]]; then # check to see if PXE vm already exists
+            echo "deploying pxe: $i"
+            vboxmanage createvm --name $vmName --register;
+            vboxmanage createhd --filename $vmName --size 8192;
+            vboxmanage storagectl $vmName --name "SATA Controller" --add sata --controller IntelAHCI
+            vboxmanage storageattach $vmName --storagectl "SATA Controller" --port 0 --device 0 --type hdd --medium $vmName.vdi
+            vboxmanage modifyvm $vmName --ostype Ubuntu --boot1 net --memory 768;
```
Comment (benbp) [inline example/bin/monorail_rack]: Should we just bump memory to 1024?

File: example/roles/repos/tasks/main.yml [hunk 4]
```diff
@@ -11,31 +11,29 @@
...
-- name: Checkout branch and set HEAD to most current
-  shell: git checkout {{ branch }}
+- name: Reset to the branch specified
+  shell: git reset --hard {{ branch }}
   when: branch is defined
   args:
     chdir: /home/vagrant/src/{{ item }}
   with_items:
     - on-core
+    - on-tasks
```
Comment (benbp) [inline example/roles/repos/tasks/main.yml]: Ah

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,3 @@
-dev-tools/.vagrant
-dev-tools/config/monorail_rack.cfg
-dev-tools/bin/pxe*
+example/.vagrant
+example/config/monorail_rack.cfg
+example/bin/pxe*
```
Comment (DavidjohnBlodgett) [inline .gitignore]: oh good catch!

File: example/bin/monorail_rack (hunk 3 above)
Comment (heckj) [inline example/bin/monorail_rack]: I tried with both, and was able to get away with 768MB, so I thought I'd try and keep it as small as possible from a quick/dev setup perspective

Comment (benbp) [inline example/bin/monorail_rack]: k

**Comment by benbp** [comment]
@DavidjohnBlodgett the casing issue of virtualboxmanage might vary depending on whether you have a case-sensitive filesystem or not.

**Comment by benbp** [comment]
:+1: after minor nit fixes.

File: example/README.md (hunk 2 above)
Comment (heckj) [inline example/README.md]: How's something like this:

This example includes a workflow that we'll use when we identify a "virtualbox"
SKU with RackHD. This workflow sets up no-op out of band management settings
for a demo and triggers an installation of CoreOS as a default flow to run
once the "virtualbox" SKU has been identified. We'll load it into our library
of workflows:

    cd ~/src/rackhd/example
    # make sure you're in the example directory to reference the sample JSON correctly

    curl -H "Content-Type: application/json" \
    -X PUT --data @samples/virtualbox_install_coreos.json \
    http://localhost:9090/api/1.1/workflows

To enable that workflow, we also need to include a SKU definition that includes
the option of another workflow to run once the SKU has been identified. This
takes advantage of the discoverySKU workflow, which will attempt to identify a
SKU and run another workflow if specified.

Comment (benbp) [inline example/README.md]: :+1:

Comment (benbp) [inline example/README.md]: "DiscoverySKU"?

**Comment by DavidjohnBlodgett** [comment]
:+1:

**Comment by rolandpoulter** [comment]
:+1:

**Pull request merged**